docs/design/coreclr/botr/data_contracts_for_coreclr.md

```diff
@@ -0,0 +1,430 @@
+# Data Contracts for CoreCLR, and the data contract stream
+
+Data contracts represent the manner in which a tool which is not the runtime can reliably understand and observe the behavior of the runtime. Contracts are defined by their documentation, and the runtime describes what contracts are applicable to understanding that runtime.
+
+## Data Contract Stream
```

Comment:
As mentioned offline, I find it confusing to call it a stream. I would call it "Contract Descriptor" or something like that.

Ideally, the contract descriptor would be physically represented as a constant memory block.